### PR TITLE
Ensure Eval Mode In Metric Functions

### PR DESCRIPTION
**Pull request by nousr**
Loss started looking like 📈 and @rom1504 noticed we weren't properly setting eval/train mode

### CODE REVIEW

**Comment by lucidrains** [comment]
@nousr ohh i see, instead of adding these, i would move https://github.com/lucidrains/DALLE2-pytorch/blob/79f388d106c0f8d9ad3b3bfe5a484b9b84a576f2/train_diffusion_prior.py#L175 to line 179 instead 🙏

**Comment by lucidrains** [comment]
it looks like dropout isn't being used in the training script, so the training vs eval mode wouldn't matter

**Comment by lucidrains** [comment]
addressed the lack of dropout issue https://github.com/lucidrains/DALLE2-pytorch/commit/a774bfefe27ceb6171ad486165739014cf1ee9f0 so dropout is beginning to look less and less useful for large transformers, but may still have a place if one is training sub 1B parameter models

**Comment by nousr** [comment]
@lucidrains gotcha gotcha

I shifted around the train toggle, and kept the eval call I added inside of ```get_cosine_similarity()```.

this was prompted by the charts in krish's lastest run https://wandb.ai/laion/diffusion-prior/runs/1blxu24j?workspace=user-nousr_laion

not sure what happened there, especially considering multi-million point runs have been completed before without this crazy jump in loss

**Comment by lucidrains** [comment]
@nousr this happens all the time with transformers training :) not that unusual

some groups even roll back before the divergence and retry at a lower learning rate 😆

**Comment by lucidrains** [comment]
@nousr looking good! merging, thank you!

**Comment by lucidrains** [comment]
@nousr if you are going to redo the experiment from scratch, do recommend turning on the `normformer` setting, which purportedly stabilizes transformer training for some practitioners
